packages/btcindexer/src/btcindexer.ts

```diff
@@ -559,37 +596,55 @@ export class Indexer {
 			return;
 		}
 
-		const blockHashes = blocksToVerify.map((r) => r.block_hash);
+		const distinctNetworks = [...new Set(blocksToVerify.map((b) => b.network))];
+		for (const network of distinctNetworks) {
+			const config = Array.from(this.packageConfigs.values()).find(
+				(c) => c.btc_network === network && c.is_active,
+			);
+			if (!config) {
+				logger.warn({
+					msg: "No active config for network verification, skipping",
+					network,
+				});
+				continue;
+			}
+
+			const client = new SuiClient(config, this.signerMnemonic);
+			const blockHashes = blocksToVerify
+				.filter((r) => r.network === network)
+				.map((r) => r.block_hash);
 
-		try {
-			const verificationResults = await this.nbtcClient.verifyBlocks(blockHashes);
-			const invalidHashes: string[] = [];
-			for (let i = 0; i < blockHashes.length; i++) {
-				if (verificationResults[i] === false) {
-					const blockHash = blockHashes[i];
-					if (blockHash) {
-						invalidHashes.push(blockHash);
+			try {
+				const verificationResults = await client.verifyBlocks(blockHashes);
+				const invalidHashes: string[] = [];
+				for (let i = 0; i < blockHashes.length; i++) {
+					if (verificationResults[i] === false) {
+						const blockHash = blockHashes[i];
+						if (blockHash) {
+							invalidHashes.push(blockHash);
+						}
 					}
 				}
-			}
 
-			if (invalidHashes.length > 0) {
-				logger.warn({
-					msg: "SPV Check: Detected reorged blocks. Updating transaction statuses.",
-					reorgedBlockHashes: invalidHashes,
-				});
-				await this.storage.updateConfirmingTxsToReorg(invalidHashes);
-			} else {
-				logger.debug({ msg: "SPV Check: All confirming blocks are valid." });
+				if (invalidHashes.length > 0) {
+					logger.warn({
+						msg: "SPV Check: Detected reorged blocks. Updating transaction statuses.",
+						reorgedBlockHashes: invalidHashes,
+					});
```

Comment:
how about moving this log to updateConfirmingTxsToReorg?

Reply:
agree, done